pkgs/servers/home-automation/evcc/default.nix

```diff
@@ -0,0 +1,129 @@
+{ pkgs
+, lib
+, stdenv
+, buildGoModule
+, fetchFromGitHub
+, mockgen
+, nixosTests
+
+# updateScript
+, coreutils
+, curl
+, jq
+, git
+, nix-prefetch-git
+, nix-update
+, nodePackages
+}:
+
+let
+  nodejs = pkgs.nodejs-12_x;
+  nodeEnv = import ../../../development/node-packages/node-env.nix {
+    inherit (pkgs) stdenv lib python2 runCommand writeTextFile;
+    inherit pkgs nodejs;
+    libtool = if stdenv.isDarwin then pkgs.darwin.cctools else null;
+  };
+  evccNodePackages = import ./node-packages.nix {
+    inherit (pkgs) fetchurl nix-gitignore stdenv lib fetchgit;
+    inherit nodeEnv;
+  };
+
+  srcJson = lib.importJSON ./src.json;
+  src = fetchFromGitHub {
+    owner = "andig";
+    repo = "evcc";
+    inherit (srcJson) rev sha256;
+  };
+
+  nodeDependencies = (evccNodePackages.shell.override (old: {
+    inherit src;
+  })).nodeDependencies;
+in
+buildGoModule rec {
+  pname = "evcc";
+  version = "0.60";
+
+  inherit src;
+
+  vendorSha256 = "0wqzpw5li3ls45411g2952d2zyyi0dmik0r448zpsxgfbpdvgiy9";
+
+  nativeBuildInputs = [
+    mockgen
+    nodejs
+  ];
+
+  buildFlagsArray = [
+    "-tags=release"
+  ];
+
+  ldflags = [
+    "-X github.com/andig/evcc/server.Version=${version}"
+    "-X github.com/andig/evcc/server.Commit=${srcJson.rev}"
+    "-s"
+    "-w"
+  ];
+
+  preBuild = ''
+    (
+      ln -s ${nodeDependencies}/lib/node_modules ./node_modules
+      export PATH="${nodeDependencies}/bin:$PATH"
+
+      npm run build
+    )
+
+    go generate ./...
+  '';
+
+  # Tries to connect to a bunch of ev charger units over network/bluetooth
+  doCheck = false;
+
+  passthru.updateScript = pkgs.writeShellScript "evcc-updater" ''
+    export PATH=${lib.makeBinPath [
+      coreutils
+      curl
+      git
+      jq
+      nix-prefetch-git
+      nix-update
+      nodePackages.node2nix
+    ]}
```

Comment:
This could be done with nix-shell.

Reply:
It could, but this cuts out the redundant evaluation that would have to be done by nix-shell.

In this case that's dwarfed by the `node2nix` invocation, so the difference is not significant. I'll assume you don't mean this as a general rule.